perception/radar_object_clustering/src/radar_object_clustering_node/radar_object_clustering_node.cpp

```diff
@@ -0,0 +1,233 @@
+    } else {
+      auto func_max_confidence = [](const DetectedObject & a, const DetectedObject & b) {
+        return a.existence_probability < b.existence_probability;
+      };
+      auto iter = std::max_element(
+        std::begin(clustered_objects), std::end(clustered_objects), func_max_confidence);
+
+      // class label
+      clustered_output_object.existence_probability = iter->existence_probability;
+      clustered_output_object.classification = iter->classification;
+
+      // kinematics
+      clustered_output_object.kinematics = iter->kinematics;
+
+      auto & pose = clustered_output_object.kinematics.pose_with_covariance.pose;
+      auto func_sum_x = [](const double & a, const DetectedObject & b) {
+        return a + b.kinematics.pose_with_covariance.pose.position.x;
+      };
+      pose.position.x =
+        std::accumulate(
+          std::begin(clustered_objects), std::end(clustered_objects), 0.0, func_sum_x) /
+        clustered_objects.size();
+      auto func_sum_y = [](const double & a, const DetectedObject & b) {
+        return a + b.kinematics.pose_with_covariance.pose.position.y;
+      };
+      pose.position.y =
+        std::accumulate(
+          std::begin(clustered_objects), std::end(clustered_objects), 0.0, func_sum_y) /
+        clustered_objects.size();
+      pose.position.z = iter->kinematics.pose_with_covariance.pose.position.z;
+
+      // Shape
+      clustered_output_object.shape = iter->shape;
+    }
+
+    // Fixed label correction
+    if (node_param_.is_fixed_label) {
+      clustered_output_object.classification.at(0).label =
+        perception_utils::toLabel(node_param_.fixed_label);
+    }
+
+    // Fixed size correction
+    if (node_param_.is_fixed_size) {
+      clustered_output_object.shape.type = autoware_auto_perception_msgs::msg::Shape::BOUNDING_BOX;
+      clustered_output_object.shape.dimensions.x = node_param_.size_x;
+      clustered_output_object.shape.dimensions.y = node_param_.size_y;
+      clustered_output_object.shape.dimensions.z = node_param_.size_z;
+    }
+    output_objects.objects.emplace_back(clustered_output_object);
+  }
+  pub_objects_->publish(output_objects);
+}
+
+bool RadarObjectClusteringNode::isSameObject(
+  const DetectedObject & object_1, const DetectedObject & object_2)
+{
+  const double angle_diff = std::abs(
+    tf2::getYaw(object_1.kinematics.pose_with_covariance.pose.orientation) -
+    tf2::getYaw(object_2.kinematics.pose_with_covariance.pose.orientation));
+  const double velocity_diff = std::abs(
+    object_1.kinematics.twist_with_covariance.twist.linear.x -
+    object_2.kinematics.twist_with_covariance.twist.linear.x);
+  const double distance = tier4_autoware_utils::calcDistance2d(
+    object_1.kinematics.pose_with_covariance.pose.position,
+    object_2.kinematics.pose_with_covariance.pose.position);
+
+  if (
+    distance < node_param_.distance_threshold && angle_diff < node_param_.angle_threshold &&
```

Comment:
May be we need to add angle diff condition ( e.g.  calc diff between 359 deg and 1deg). 

```py
# normalization
while angle_diff > pi:
     angle_diff -= pi

# angle condition
angle_condition = angle_diff < th  or  angle_diff > pi - th 
```

Reply:
fixed at [c6bfcad](https://github.com/autowarefoundation/autoware.universe/pull/4144/commits/c6bfcad2146e1d4d69a8197bbb6e99824a65a166) using [util function](https://github.com/autowarefoundation/autoware.universe/blob/feat/radar_object_clustering/common/tier4_autoware_utils/include/tier4_autoware_utils/math/normalization.hpp#L36) instead of your suggestion.

Reply:
LGTM